include/MidiEvent.h

```diff
@@ -37,34 +37,38 @@ class MidiEvent
 				int8_t channel = 0,
 				int16_t param1 = 0,
 				int16_t param2 = 0,
-				const void* sourcePort = NULL ) :
+				const void* sourcePort = NULL,
+				bool fromHardware = true ) :
 		m_type( type ),
 		m_metaEvent( MidiMetaInvalid ),
 		m_channel( channel ),
 		m_sysExData( NULL ),
-		m_sourcePort( sourcePort )
+		m_sourcePort( sourcePort ),
+		m_fromHardware( fromHardware )
 	{
 		m_data.m_param[0] = param1;
 		m_data.m_param[1] = param2;
 	}
 
-	MidiEvent( MidiEventTypes type, const char* sysExData, int dataLen ) :
+	MidiEvent( MidiEventTypes type, const char* sysExData, int dataLen, bool fromHardware = true ) :
 		m_type( type ),
 		m_metaEvent( MidiMetaInvalid ),
 		m_channel( 0 ),
 		m_sysExData( sysExData ),
-		m_sourcePort( NULL )
+		m_sourcePort( NULL ),
+		m_fromHardware( fromHardware )
 	{
 		m_data.m_sysExDataLen = dataLen;
 	}
 
-	MidiEvent( const MidiEvent& other ) :
+	MidiEvent( const MidiEvent& other, bool fromHardware = true ) :
```

Comment:
Do you really need `fromHardware` for the copy constructor?

Reply:
Sorry about that, I'm changing it so it just copies the value from the other event!